### PR TITLE
Development/queens bugfix

### PR DESCRIPTION
**Pull request by tsimchev**

### CODE REVIEW

File: networking_nsxv3/common/scheduling.py [hunk 1]
```diff
@@ -44,8 +44,8 @@ def __enter__(self):
             sleeptime = run_time - self.schedule[offset] + self.limit
             if self.callback:
                 eventlet.spawn(self.callback, sleeptime)
-            eventlet.greenthread.sleep(sleeptime)
             run_time = self.schedule[offset] + self.limit
```
Comment (joker-official) [inline networking_nsxv3/common/scheduling.py]: `eventlet.spawn()` calls into eventlet, too, so there might be greenthread-scheduling involved. Could we move the `run_time` computation 2 more lines up?

File: networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py [hunk 2]
```diff
@@ -33,6 +33,10 @@
 )
 
 
+class HttpUnsuccessfulExceptoin(Exception):
```
Comment (joker-official) [inline networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py]: typo `Exceptoin`

File: networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py [hunk 3]
```diff
@@ -56,22 +60,34 @@ def decorator(self, *args, **kwargs):
...
+                                resp.status_code, resp.reason))
+
+                    if resp.status_code > 400 and resp.status_code < 500:
                         raise Unauthorized(resp.content)
+
+                    if resp.status_code >= 300:
+                        msg = "HTTP Response URL={} Code={} Reason={} \
+                            Content={}".format(resp.url, resp.status_code,
+                                               resp.reason, resp.content)
+                        raise HttpUnsuccessfulExceptoin(msg)
```
Comment (joker-official) [inline networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py]: same typo as above

File: networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py [hunk 4]
```diff
@@ -211,7 +227,14 @@ def _query(self, resource_type, key, ands=[], ors=[], dsl="",
...
         resp = self._post(path="/nsxapi/rpc/call/SearchFacade", data=data)
-        return json.loads(resp.content)["result"]["results"]
+        content = json.loads(resp.content)
+
+        if content.get("result") is None\
+                or content.get("result").get("results") is None:
+            raise HttpUnsuccessfulExceptoin(
+                "Expected 'results' in payload={}".format(resp.content))
+
+        return content.get("result").get("results")
```
Comment (joker-official) [inline networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py]: Why the `get` here? Doesn't the check 4 lines up cover this?

Comment (tsimchev) [reply networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_client.py]: Yes, the above check covers this.

**Pull request merged**